### ISSUE TITLE
Can't search children tags on collections module

### ISSUE THREAD

**Issue by Nilvus** [darktable-org/darktable#12471]
**Describe the bug/issue**

I can't search a children tag on collections module. @phweyland: as you work on that, maybe you will have some ideas to fix that.

**To Reproduce**
1. Be sure to have no other filter that will go against your search (having all images displayed)
2. Select tag filter on collections module
3. Type a tag which is a children, not first parent
4. See that collections list became empty.

**Expected behavior**

When typing any tag, independently of hierarchy, the tag should be displayed to allow display related images.

Hard to know when this issue was introduced. I didn't use that search since months but I remember that this was working, at least last year.

**Platform**

* darktable version : actual master
* Linux - Distro    : Debian Sid

**Comment by elstoc** [comment]
I think you need to use a wildcard to search child tags. I don't use tags myself but if I want to search for all cr2 files using the `darktable|format|cr2` tag, I have to use `%cr2` rather than just `cr2`.

**Comment by Nilvus** [comment]
Indeed, using wildcard works. I don't remember using it before. This is indeed really inconsistent. So I hope to see this improved. I think that this shouldn't need wildcard for such search on tags. This is also not obvious to use wildcard to search children tags.

**Comment by phweyland** [comment]
> This is indeed really inconsistent

On the contrary, the current behavior has been adopted to ensure consistency between the different hierarchical tables (folders, tags,...)
Obviously, some may prefer the other way, but in any case, collection should show a consistent behavior.

**Comment by Nilvus** [comment]
I understand the idea to add % as the tooltip explains, for a joker. So for searching part of whatever we want but if I search a specific one (here tag) that I know the entire tag, adding 'joker' is not obvious.

**Comment by phweyland** [comment]
The tagging module does not require to use the wildcard, which is a more friendly usage (for dummy). But to get a specific tag the user has to enter the full tag.
The global text filter behaves the same by default (automatic wildcard) , but has a special character to disable the wildcard. But I find the code horrible (that's not the natural sql way). 
The collection module could behave like this as well, provided that it remains coherent internally (folders, tags,...) .

**Comment by Nilvus** [comment]
I never talk about tagging module here. I talk about searching on collections module.

**Comment by EC1000** [comment]
> On the contrary, the current behavior has been adopted to ensure consistency between the different hierarchical tables (folders, tags,...)
> Obviously, some may prefer the other way, but in any case, collection should show a consistent behavior.

I do not use folder often (which indeed works as you say @phweyland), however I use film roll a lot, and here the search works without a wildcard (please do not change that for the sake of consistency, I would regret writing this :smile:). 

However, it seems to me there are not two but three different behaviors: 
1. you need to enter a wildcard to reduce the collection list and to collect images (e.g. tags or folders)
2. you don't need to enter a wildcard to reduce the collection list, but you need one to collect images (e.g. film rolls)
3. you don't need to enter a wildcard neither to reduce the collection list, nor to collect images (e.g. description)

So for the second behavior, there is no automatic wildcard to actually collect image. If you enter: `something` it will reduce the list to show you `this day when I did something` and `this something I photographed`, but if you hit Enter the collection will be empty. 

As Nilvus, what I would like would be the second behavior also for tags (and it would probably make sens for folders and others that use the first behavior, and for consistency, but I personnally use them less so I have not been impacted).
